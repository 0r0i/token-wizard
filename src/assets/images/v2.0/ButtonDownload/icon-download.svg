<svg xmlns="http://www.w3.org/2000/svg" width="21" height="24"><path d="M1 17.5v3.3c0 1.215.945 2.2 2.111 2.2H17.89c1.165 0 2.11-.985 2.11-2.2v-3.3M6.278 12l4.222 4.4 4.222-4.4M10.5 1v15.4" fill="none" fill-rule="evenodd" stroke="#FFF" stroke-linecap="round" stroke-linejoin="round" stroke-width="1.5"/></svg>