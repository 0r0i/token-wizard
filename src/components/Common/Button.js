import React from 'react'

export const Button = ({ buttonText = '', disabled = false, extraClassName = '', onClick, type = 'button' }) => (
  <button className={`sw-Button ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    {buttonText}
  </button>
)
