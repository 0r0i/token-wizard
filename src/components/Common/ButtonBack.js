import React from 'react'

export const ButtonBack = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonBack ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    Back
  </button>
)
