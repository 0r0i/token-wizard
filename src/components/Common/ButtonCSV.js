import React from 'react'

export const ButtonCSV = ({ onClick, text, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonCSV ${extraClassName}`} disabled={disabled} onClick={onClick} type="button">
    {text}
  </button>
)
