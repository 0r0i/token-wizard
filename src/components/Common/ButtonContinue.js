import React from 'react'

export const ButtonContinue = ({
  buttonText = 'Continue',
  disabled = false,
  extraClassName = '',
  onClick,
  type = 'button'
}) => (
  <button className={`sw-ButtonContinue ${extraClassName}`} disabled={disabled} onClick={onClick} type={type}>
    <span className={`sw-ButtonContinue_Text`}>{buttonText}</span>
  </button>
)
