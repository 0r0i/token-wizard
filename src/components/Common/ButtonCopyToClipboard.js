import React from 'react'

export const ButtonCopyToClipboard = ({ disabled, extraClassName, value }) => (
  <button
    className={`sw-ButtonCopyToClipboard ${extraClassName ? extraClassName : ''}`}
    data-clipboard-action="copy"
    data-clipboard-text={value}
    disabled={disabled}
    type="button"
  />
)
