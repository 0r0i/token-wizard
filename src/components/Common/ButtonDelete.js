import React from 'react'

export const ButtonDelete = ({ onClick, extraClassName = '', disabled }) => (
  <button className={`sw-ButtonDelete ${extraClassName}`} disabled={disabled} onClick={onClick} type="button" />
)
