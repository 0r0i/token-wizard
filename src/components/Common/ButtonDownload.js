import React from 'react'

export const ButtonDownload = ({ disabled, onClick, extraClass = '' }) => (
  <button className={`sw-ButtonDownload ${extraClass}`} disabled={disabled} onClick={onClick} type="button">
    Download
  </button>
)
