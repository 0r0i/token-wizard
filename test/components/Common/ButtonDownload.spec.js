import React from 'react'
import render from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-15'
import { configure, shallow } from 'enzyme'
import { ButtonDownload } from '../../../src/components/Common/ButtonDownload'

configure({ adapter: new Adapter() })

describe(`ButtonDownload`, () => {
  it(`should be a <button> HTML tag`, () => {
    const wrapper = shallow(<ButtonDownload />)

    expect(wrapper.type()).toBe('button')
  })

  it(`its type attribute should be 'button'`, () => {
    const wrapper = shallow(<ButtonDownload />)

    expect(wrapper.props().type).toBe('button')
  })

  it(`should render ButtonDownload component`, () => {
    // Given
    const component = render.create(<ButtonDownload onClick={jest.fn()} />)

    // When
    const tree = component.toJSON()

    // Then
    expect(tree).toMatchSnapshot()
  })
})
