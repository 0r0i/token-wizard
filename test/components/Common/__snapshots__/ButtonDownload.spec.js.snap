// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonDownload should render ButtonDownload component 1`] = `
<button
  className="sw-ButtonDownload "
  onClick={[MockFunction]}
  type="button"
>
  Download
</button>
`;
